sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQos.java

```diff
@@ -200,11 +200,10 @@ interface RpcWriteAttempt extends RpcAttempt {
      * provided {@code instant}.
      *
      * @param instant The intended start time of the next rpc
-     * @param <T> The type which will be sent in the request
      * @param <ElementT> The {@link Element} type which the returned buffer will contain
      * @return a new {@link FlushBuffer} which queued messages can be staged to before final flush
      */
-    <T, ElementT extends Element<T>> FlushBuffer<ElementT> newFlushBuffer(Instant instant);
+    <ElementT extends Element<?>> FlushBuffer<ElementT> newFlushBuffer(Instant instant);
```

Comment:
This seems like a regression in functionality. Now the type cannot assure that it is a `FlushBuffer<Element<T>>`

Reply:
Admit that it is a compile time regression. Not matter at runtime as generics are erased.

RpcWriteAttempt / RpcWriteAttempt.FlushBuffer / RpcWriteAttempt.Element are package private visible interfaces so user won't feel it